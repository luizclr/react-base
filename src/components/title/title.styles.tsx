import { getColorByTheme, getFontSize } from "react-styled-guide";
import styled from "styled-components";

export const Title = styled.h1`
  font-size: ${getFontSize(({ xxl }) => xxl)};
  color: ${getColorByTheme(
    ({ neutral }) => neutral.darkest,
    ({ neutral }) => neutral.lightest
  )};
`;
