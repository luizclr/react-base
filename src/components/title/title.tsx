import { PropsWithChildren } from "react";

import { Title as BaseTitle } from "~/components/title/title.styles";

export const Title: React.FC<PropsWithChildren> = ({ children }: PropsWithChildren) => (
  <BaseTitle>{children}</BaseTitle>
);
