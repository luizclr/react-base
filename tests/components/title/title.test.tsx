import { faker } from "@faker-js/faker";
import { StyleGuide, ThemeTypes } from "react-styled-guide";

import { Title } from "~/components/title/title";
import initialStyleGuide from "~/style-guide";

import { render } from "#/test-utils/render";

const makeSut = (): {
  sut: HTMLElement;
  styleGuide: StyleGuide;
} => {
  const titleText = faker.word.sample();
  const { getByText } = render(<Title>{titleText}</Title>);

  const sut = getByText(titleText);

  const styleGuide = initialStyleGuide.styleGuide;

  return { sut, styleGuide };
};

describe("<Title />", () => {
  it("should be able to mount <Title /> component", () => {
    // given
    const { asFragment } = render(<Title>Hello world</Title>);

    // when
    const sut = asFragment();

    // then
    expect(sut).toMatchSnapshot();
  });

  it("should has dark color on light theme", () => {
    // given
    localStorage.setItem("theme", ThemeTypes.light);
    const { sut, styleGuide } = makeSut();
    const { darkest } = styleGuide.colors.neutral;

    // then
    expect(sut).toHaveStyleRule("color", darkest);
  });

  it("should has light color on dark theme", () => {
    // given
    localStorage.setItem("theme", ThemeTypes.dark);
    const { sut, styleGuide } = makeSut();
    const { lightest } = styleGuide.colors.neutral;

    // then
    expect(sut).toHaveStyleRule("color", lightest);
  });
});
