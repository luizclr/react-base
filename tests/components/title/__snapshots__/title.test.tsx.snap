// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should be able to mount <Title /> component 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 32pt;
  color: #1c1c1c;
}

<h1
    class="c0"
  >
    Hello world
  </h1>
</DocumentFragment>
`;
